src/main/java/com/github/tomakehurst/wiremock/extension/ExtensionFactory.java

```diff
@@ -15,8 +15,15 @@
  */
 package com.github.tomakehurst.wiremock.extension;
 
+import static java.util.Collections.emptyList;
+
 import java.util.List;
 
 public interface ExtensionFactory {
+
   List<Extension> create(WireMockServices services);
+
+  default List<Extension> createForClient() {
```

Comment:
I am slightly concerned about this, as it makes it a little more difficult to properly build an `ExtensionFactory`. I am also curious if anyone would ever require `WireMockServices` to build a `ContentPatternExtension` (some really bad code that looks up the `getContentPatternClass` from an option or file).